flocker/node/test/test_script.py

```diff
@@ -229,6 +234,58 @@ def test_initialized(self):
             ),
         )
 
+    def test_logging(self):
+        """
+        Logging is setup by a logging stanza.
+        """
+        # Save and restore root logger state
+        def restore_logger(logger, level, handlers):
+            logger.setLevel(level)
+            for handler in logger.handlers[:]:
+                logger.removeHandler(handler)
+            for handler in handlers:
+                logger.addHandler(handler)
+
+        logger = logging.getLogger()
+        self.addCleanup(
+            restore_logger, logger, logger.getEffectiveLevel(),
+            logger.handlers[:]
+        )
+
+        # Setup an AgentService with a logging stanza
+        host = b"192.0.2.13"
+        port = 2314
+        name = u"from_config-test"
+        logfile = self.make_temporary_directory().child('logfile')
```

Comment:
OK, looked at the testtools code again, and can see that each call to `make_*` creates a new directory for the contents.